lib/lightning/extensions/runtime_limiter.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Lightning.Extensions.RuntimeLimiter do
+  @moduledoc """
+  Adapter to call the extension for limiting Runtime workloads.
+  """
+  @behaviour Lightning.Extensions.RuntimeLimiting
+
+  import Lightning.Extensions.AdapterHelper
+
+  alias Lightning.Extensions.RuntimeLimiting
+  alias Lightning.Extensions.RuntimeLimiting.Action
+  alias Lightning.Extensions.RuntimeLimiting.Context
+
+  @type message :: %{
+          position: atom(),
+          function: fun(),
+          attrs: Keyword.t()
+        }
+
+  @spec check_limits(Context.t()) ::
+          :ok | {:error, RuntimeLimiting.limit_error(), message()}
+
+  def check_limits(context) do
+    adapter().check_limits(context)
+  end
```

Comment:
Should these functions not use `@impl true` instead of having their own specs?

I see the contract for `check_limits` is different from the behaviour, dialyzer doesn't pick it up because of the specs, a little puzzling because I would assume it would moan about either being different or think the function is missing.

Removing the `@behaviour` has no effect on dialyzer.

Reply:
I remember a case where dialyzer haven't checked properly only with `@impl true` but it's been a while. Let's see.